src/SFML/Network/Http.cpp

```diff
@@ -27,26 +27,13 @@
 ////////////////////////////////////////////////////////////
 #include <SFML/Network/Http.hpp>
 #include <SFML/System/Err.hpp>
-#include <cctype>
+#include <SFML/System/String.hpp>
```

Comment:
`String.hpp` is one of the most expensive SFML headers to compile. We are adding it here with the only purpose of exposing `toLower` -- I am not sure that is a good idea in the context of compilation times.

Reply:
I agree. Chances are there are more such utilities like `sf::toLower()`, and those could maybe combined in an own header `System/Utils.hpp` (if dependency-heavy, with separate `.cpp` file)?

Reply:
@vittorioromeo I totally agree. I originally wanted to put this in a private header somewhere in src/ since this is only needed in .cpp files. However, I saw no internal generic utilities file/library and I saw that many files included a `priv::` namespace in public headers for things like this so I copied that pattern.

@Bromeon I think a Utils.[hpp|cpp] is a great idea. There ought to be a central location for common implementation details like this so we can encourage better code reuse.

Reply:
The only problem I see is that such "utils" tend to aggregate all kinds of unrelated functionality, which often comes with various different `#include`s, ultimately increasing compilation time for each of the client files.

For example, I was thinking this might be a place for the min/max functionality used in `Rect` and `Vector2/3`, but that would mean those need to include `<string>`, which is a no-no.

We can still start with `Utils`, and rename it as we see fit.

Reply:
I honestly would also be fine with something like `Utils/ToLower.hpp`, and have `Utils` as a folder.

Reply:
Also a possibility. In case of small dependencies (here `<cctype>` which might already be included by `<string>`), we can also abandon the .cpp file entirely.

Reply:
What if we put private headers in src/? That way we can avoid `priv::` and `Priv/` and have a much stronger guarantee that no user will depend on these utilities.